### ISSUE TITLE
can't read "lib_specifications": no such variable

### ISSUE THREAD

**Issue by horvathp** [benjmarshall/hlsclt#7]
Hello,

I'm a happy user of hlsclt. However, I tried to port a project of mine to a bigger part,
part_name = "xczu9eg-ffvc900-2i"
and now the build csim (and every other task) fails with

```
can't read "lib_specifications": no such variable
    while executing
"source [lindex $::argv 1] "
    ("uplevel" body line 1)
    invoked from within
"uplevel \#0 { source [lindex $::argv 1] } "
```
I can't make anything of this error. Developing under the GUI works for this part, too (no license issues). Google didn't help. Strangely, a 442-byte binary file named 'c' remains in the work directory after the unsuccessful run.
HLS 2017.4 here.

Best,
Peter

**Comment by horvathp** [comment]
I noted that the HLS GUI knows this particular part as 
xczu9eg-ffvc900-2-i
but this form does not gets recognized, 
`ERROR: [HLS 200-101] 'set_part': Argument 'device_specification' requires a value!
`

**Comment by benjmarshall** [comment]
Hi Peter,

Can you upload a zip of your project folder? Or if your source is sensitive at least a copy of the generated run.tcl?

Thanks, 
Ben

**Comment by horvathp** [comment]
The generated tcl:
```
open_project proj_fake_axis_loopback
set_top fake_axis_loopback
add_files ./fake_axis_loopback.cpp
add_files ./fake_axis_loopback.h
add_files -tb ./fake_axis_loopback_tb.cpp
open_solution "solution1"
set_part \{xczu9eg-ffvc900-2-i\}
create_clock -period 4 -name default
csim_design -clean
exit
```

I tried this with a very basic block, so if it is of concern, I can share all the 20 lines of code but I think it does not reach the point where sources would be looked at. It works well under the GUI.

**Comment by horvathp** [comment]
The GUI generates the part name without backslash-es:
`set_part {xczu9eg-ffvc900-2-i}
`

**Comment by benjmarshall** [comment]
Thanks, I'll try and take a look. Can you let me know your platform details? OS, tool versions etc?

**Comment by horvathp** [comment]
hlsclt, version 1.0.0.a4
OS: Ubuntu 16.04, Vivado 2017.4
Python 2.7.12 & 3.5.2 (no difference wrt to Python version)

As I said, it works well for an xc7something part, so maybe the multiple dashes in the part name have to do with the issue.

**Comment by benjmarshall** [comment]
Ok this definetly seems to be an issue with that specific part string, and something to do with the backslashes. I can push an update which removes the slashes from generated run tcl which should fix!

**Comment by horvathp** [comment]
I tested it, and without the braces, it works. So if there is no need for the braces otherwise, I'd suggest removing them.
Thank you very much for looking into this issue.

**Comment by benjmarshall** [comment]
Fixed in release 1.0.0.a5.
